KeenClient/KeenClient.m

```diff
@@ -291,78 +228,140 @@ + (BOOL)validateKey:(NSString *)key {
     return [KeenClient validateProjectID:key];
 }
 
-- (id)initWithProjectID:(NSString *)projectID andWriteKey:(NSString *)writeKey andReadKey:(NSString *)readKey {
+- (instancetype)initWithNetwork:(KIONetwork*)network
+                       andStore:(KIODBStore*)store
+                    andUploader:(KIOUploader*)uploader {
+    self = [super init];
+
+    if (nil != self) {
+        // log the current version number
+        if ([KeenClient isLoggingEnabled]) {
```

Comment:
Why do we check isLoggingEnabled here but not elsewhere. I thought KCLog() performed that check internally. PR #188 changes this a bit, but the check for logging enabled is still handled somewhere down in the logger I think, so it seems like either case this check can go away.